addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -245,8 +244,9 @@ export class PaymentScreen extends Component {
             );
             return;
         }
-
-        pLine.setAmount(pLine.getAmount() - (tip || 0) + parseFloat(newTip));
+        const tipDifference = parseFloat(newTip) - (tip || 0);
+        const tipToAdd = change < 0 ? tipDifference : Math.max(0, tipDifference - change);
```

Comment:
```suggestion
        const tipToAdd = change <= 0 ? tipDifference : Math.max(0, tipDifference - change);
```
Shouldn't it be <= 0 ? Because if change is 0 we can just add the tipDifference I think. The only moment we want to make the other calculation is if there is already some change.